### DISCUSSION TITLE
Fetch user data in router, before component initialization

### DISCUSSION THREAD

**Discussion by filipbekic01** [DamianOsipiuk/vue-query#255] [Q&A] [answered]
Hello there,

I want to fetch user data using vue-query before vue-router enters component.

When I try to use query it says hooks can't be used before component is instanciated.

Any suggestions?
Cheers

**Comment by DamianOsipiuk** [comment, marked as answer]
I think you have two options:
- [const data = await queryClient.fetchQuery(queryKey, queryFn)](https://tanstack.com/query/v4/docs/reference/QueryClient#queryclientfetchquery)
- suspense
  https://vuejs.org/guide/built-ins/suspense.html
  https://github.com/DamianOsipiuk/vue-query/tree/main/examples/suspense/src

Depending on the data type and use-case

**Reply by filipbekic01** [reply]
Query client is what we're looking into right now. Thanks for help!
